server/lib/adapters/snowflake.js

```diff
@@ -0,0 +1,177 @@
+/**
+ * Module dependencies.
+ */
+import snowflake from "snowflake-sdk";
+import Promise from "bluebird";
+import SequelizeUtils from "sequelize/lib/utils";
+import validateResultColumns from "./validate-result-columns";
+import _ from "lodash";
+
+/**
+ * SnowFlake adapter.
+ */
+
+/**
+ * Open a new connection.
+ *
+ * @param {Object} settings The ship settings.
+ *
+ */
+export function openConnection(settings) {
+  // Must specify region even though it's supposed to be deprecated
+  // Known issue with nodejs sdk (today is Jan 15 2019)
+  // https://stackoverflow.com/questions/54129786/snowflake-dw-conncetion-with-node-js-driver
+  const conf = {
+    account: settings.db_account,
+    region: settings.db_region,
+    username: settings.db_user,
+    password: settings.db_password,
+    database: settings.db_name
+  };
+
+  // Cannot create connection if these things are blank
+  // snowflake actually does validation with this, the problem is that we
+  // call openConnection from the constructor... before we validate the input
+  // so there are cases where it blows up early... not what a constructor should be doing
+  if (
+    _.isEmpty(conf.account)
+    || _.isEmpty(conf.region)
+    || _.isEmpty(conf.username)
+    || _.isEmpty(conf.password)
+    || _.isEmpty(conf.database)
+  ) {
```

Comment:
I think this is very sane approach you took. My only question would be what does it mean "breaks really badly"? Does those checks prevent that in all cases?

Reply:
"breaks really badly" means that it can throw errors which completely break the connector and don't let it start again.  But could also mean that it just throws an error if something like username isn't there.

This pattern of "openConnection" before we validate the parameters should be something that we change in the future anyway.  for some reason the other connectors are ok with it, and don't actually try to open the connection until much later.